harry-core/src/harry/visitors/MutatingVisitor.java

```diff
@@ -116,21 +163,23 @@ public void afterLts(long lts, long pd)
             statements.clear();
             bindings.clear();
 
-            executeWithRetries(lts, pd, new CompiledStatement(query, bindingsArray));
+            CompiledStatement compiledStatement = new CompiledStatement(query, bindingsArray);
+            executeWithRetries(lts, pd, compiledStatement);
             tracker.endModification(lts);
+            hadVisibleVisit = false;
         }
 
         @Override
-        public void operation(long lts, long pd, long cd, long opId, OpSelectors.OperationKind opType)
+        public void operation(Operation operation)
         {
-            CompiledStatement statement = operationInternal(lts, pd, cd, opId, opType);
-            statements.add(statement.cql());
-            Collections.addAll(bindings, statement.bindings());
+            hadVisibleVisit = operation.opKind() != OpSelectors.OperationKind.DELETE_PARTITION;
```

Comment:
Aren't delete range / slice also not visible?

Reply:
They are visible since if there is a partition liveness marker, partition's static column is going to survive. Adding a comment!